Fix permission issues and refactor postgresql_dump helper script

sthirugn opened the pull request:
Fixes #159

johnpmitsch commented on helpers/postgresql_dump.sh:
```diff
@@ -1,12 +1,34 @@
+# Run katello-backup
 katello-backup $backup_dir
+echo "pgsql_data.tar.gz is not required for rhel6 to 7 migration..."
+rm -f $backup_dir/katello-backup*/pgsql_data.tar.gz
```
The `katello-backup*` makes me nervous as the user may have multiple katello-backups and this will remove pgsql_data.tar.gz from all of them

sthirugn replied:
your concern is valid, do you have an other suggestion?

johnpmitsch replied:
@sthirugn the only thing I can think of is to grab the folder name from the second line of output here:
```
Starting backup: 2017-03-29 19:20:12 +0000
Creating backup folder /backup/katello-backup-2017-03-29T19:20:12+00:00
``` 
but then there is the issue of the special characters...

I'll think about it some more though

Maybe we can just print a reminder to the user to not copy the psql_data.tar.gz file  over when doing a RHEL migration

sthirugn replied:
I am adding an explicit note to not copy the pgsql file.

johnpmitsch replied:
@sthirugn if you want to include it, it looks like this will work
```
[root@sat-628-rhel7 /]# backup_dir=$(cat /tmp/backup_text.txt | awk '{w=$9} END{print w}')
[root@sat-628-rhel7 /]# cd $backup_dir
[root@sat-628-rhel7 katello-backup-2017-03-29T19:20:12+00:00]# 
```
That is just from the saved output from the file so you will have to replace that cat part and probably change the variable name since its  already used by you

johnpmitsch replied:
@sthirugn this is working for me from the command line:
```
[root@sat-snap-rhel7 ~]# backup_subdir=$(katello-backup /backup | awk '{w=$9} END{print w}')
[root@sat-snap-rhel7 ~]# cd $backup_subdir
[root@sat-snap-rhel7 katello-backup-2017-03-29T19:41:43+00:00]#

sthirugn replied:
@johnpmitsch this works but relying on the word position of 9 in awk command may not be a reliable way to run in production.  I would rather leave it with a text warning.

sthirugn replied:
note that when katello-backup gets updated this position of the folder name can change and it is not safe.

johnpmitsch replied:
@sthirugn ya, I was thinking the same thing, I am ok with the warning, I am ok with the warning

johnpmitsch commented on helpers/postgresql_dump.sh:
```diff
@@ -29,6 +27,8 @@ katello-service start
+echo "config_files.tar.gz, mongo_data.tar.gz, pulp_data.tar,"\
      "candlepin.dump, foreman.dump"
+echo "Note: Do not copy the generated pg_sql_data.tar.gz as this is not"\
+     "required for migration process."
```
@sthirugn Please specify `RHEL 6 to 7 migration` here as `migration` is a bit ambiguous

johnpmitsch commented:
@sthirugn it looks fine other than my one comment - I'll try it out on a machine tomorrow

sthirugn commented:
@johnpmitsch thanks, updated your comment

johnpmitsch commented on helpers/postgresql_dump.sh:
```diff
@@ -1,12 +1,34 @@
+echo "config_files.tar.gz, mongo_data.tar.gz, pulp_data.tar,"\
+     "candlepin.dump, foreman.dump"
+echo "Note: Do not copy the generated pg_sql_data.tar.gz as this is not"\
+     "required for RHEL6 to 7 migration process."
```
`pg_sql_data.tar.gz` should be `pgsql_data.tar.gz`

johnpmitsch replied:
The user may not know where the backup files are located as the location displayed in the katello-backup output gets lost in the churn

sthirugn replied:
valid concern.  User can use a new backup folder everytime.  This can be tracked and improved in https://github.com/RedHatSatellite/satellite-clone/issues/164

johnpmitsch commented:
@sthirugn The script ran correctly and without errors :+1:

After running the script, I am left with this file structure
```
/backup/
├── candlepin.dump
├── foreman.dump
└── katello-backup-2017-03-30T14:41:06+00:00
    ├── config_files.tar.gz
    ├── metadata
    ├── mongo_data.tar.gz
    ├── pgsql_data.tar.gz
    └── pulp_data.tar
```
The necessary files are split up into two different directories, what is expected of the user? Should they know that the candlepin.dump and foreman.dump files are associated with `katello-backup-2017-03-30T14:41:06+00:00`? I would also think that these files would be overwritten for each backup made.

This is making me reconsider the awk and find the filename from the output approach. The files could be moved to the correct backup folder, and the pgsql_data file can be removed. We can also display where the backup file is located at the end of the script output. Relying on the position of text output is not ideal, but I dont forsee that line changing in 6.2. We will also know about any upstream changes ahead of time and can handle those appropriately.

I'm open to alternative suggestions

johnpmitsch commented:
@sthirugn 
I think we can clean up this [prerequisites section](https://github.com/RedHatSatellite/satellite-clone/blob/master/docs/satellite-clone.md#prerequisites) of the README. My thought is:
- rename this script to `rhel_6_to_7_migration_backup.sh` or something similar, since its explicitly used for generation backups for RHEL migrations
- Remove the additional notes section of that section, and make a "sub-bullet" under RHEL 6 to 7 migration scenario that tells the user to run this script to generate the appropriate backup
- The pulp data file being optional is made clear by the `(optional)` part in the list, we can either enhance that or leave it as is, but IMO no need to mention it again.

I bring this up as @pgagne mentioned he was a bit confused by that section, maybe he can comment with any clarifications as to what confused him

sthirugn commented:
@johnpmitsch thanks

- I addressed your comments mentioned in https://github.com/RedHatSatellite/satellite-clone/pull/161#issuecomment-290440629 - great suggestions btw
- For https://github.com/RedHatSatellite/satellite-clone/pull/161#issuecomment-290437934 -This PR is originally created for addressing permission issues in the script and it is already addressed.  katello-backup creating a new folder is a new issue introduced with 6.2.8 and needs to be addressed in another PR imho and should not block this PR.  I created a separate issue https://github.com/RedHatSatellite/satellite-clone/issues/164 to track this.

sthirugn commented:
@johnpmitsch as per the irc conv, the new output will look like:

```sh
Backup is created in /backup folder. Now copy the generated files to the destination host:
config_files.tar.gz, mongo_data.tar.gz, pulp_data.tar, candlepin.dump, foreman.dump
Note: Do not copy the generated pgsql_data.tar.gz as this is not required for RHEL6 to 7 migration process.
```

sthirugn commented:
Ty @johnpmitsch
